src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -127,4 +142,20 @@ class WidgetController(
     private fun setEiDClientURL(url: String): Pair<String, String> {
         return "eidClientURL" to url
     }
+
+    /*
+       Since user agent will return version in String, we need to properly convert the string to integer,
+       otherwise we cannot perform logical operations such as <= or >=
+       If major version returns empty or somehow the user agent parser cannot parse the version properly,
+       we will show to users the widget
+    */
+    private fun safelyGetMajorOSVersion(majorVersion: String?): Int {
+        val fallbackOSVersion = 20
```

Comment:
Magic numbers aren't a good pattern, if they don't stand for what they mean (i.e. 20 would be only okay here, if `majorVersion` is always empty exactly on version 20).
Therefore, rather return `null` here (and `Int?` as return type) and e.g. add another check `!= null` in rows 66 and 67.

Reply:
yes, I agree! I will try the return type! thanks!

Reply:
done! please refer to this commit: https://github.com/digitalservicebund/useid-backend-service/pull/129/commits/0b5da7389bfd1560a4cf878edf554300b8b92a68